code/SConstruct

```diff
@@ -177,24 +177,27 @@ if flag_debug:
     env.Append(CXXFLAGS='-fno-optimize-sibling-calls')
     # keep the frame pointer for better debugging
     env.Append(CXXFLAGS='-fno-omit-frame-pointer')
-    # undefined behavior sanitizer, apple clang unsupported
-    if clang_version and clang_version >= (3, 9):
-        env.Append(CXXFLAGS='-fsanitize=undefined')
-        env.Append(extra_libs=['ubsan'])
-    # memory error sanitizer
-    if flag_sanitize:
+    if flag_sanitize:   
+        # memory error sanitizer
         env.Append(CXXFLAGS='-fsanitize=address')
```

Comment:
Can you have `-fsanitize=address` without `-lasan`?

Reply:
This is how it was written before my changes, I just rearranged the structure to reduce some repetition and simplify the `if` flow. Or at least that was my intention.

From the "native clang doesn't need explicit asan lib" comment, I assumed having the `env.Append(CXXFLAGS='-fsanitize=address')` line on its own was related to Apple clang support. I don't use Apple myself so have no way to test it.

My thought process was that if someone were compiling using native Apple clang with flag_sanitize = true, they'd get the `env.Append(CXXFLAGS='-fsanitize=address')` line by default, wouldn't need the `env.Append(extra_libs=['asan'])` line per the comment, and wouldn't be compatible with the rest of the stuff inside that branch anyway.

Anyone compiling from a non-Apple OS would get the same `env.Append(CXXFLAGS='-fsanitize=address')` line by default, and would be guaranteed to have either gcc_version or clang_version and clang_version >= (3, 9) so they'd get the `env.Append(extra_libs=['asan'])` line added in for sure plus whatever other stuff evaluated true.

Any preference on how you'd like me to fix it? Maybe just move the `env.Append(CXXFLAGS='-fsanitize=address')` inside the first `if` statement with the other Appends?

Reply:
Nobody's tried to run Sneezy on a Mac for many years (to my knowledge) and whoever would try that would probably use Docker anyway, so maybe it's not very important to preserve Mac compatibility.